src/core/postNote.ts

```diff
@@ -109,25 +158,67 @@ export class PostNote {
             while (images.length > 0) {
                 images[0].parentNode.removeChild(images[0]);
             }
-            this.note['body_html'] = dom.body.innerHTML;
+
+            const htmlToText = dom.body.innerHTML
+                .replace(/<\/?(table|tr|td|th)\b[^>]*>/gi, '\n\n')
+                .replace(/\r?\n/g, '\u0001')
+
+                // convert linebreak placeholders back to newlines
+                .replace(/\u0001/g, '\n');
+
+            this.note['body_html'] = htmlToText + attachmentsLinks;
+            this.note['markup_language'] = 1;
         }
     }
 
-    async addInlineAttachments(attachmentsProp: AttachmentProperties[]) {
+    async addInlineAttachments(): Promise<void> {
         const domParser = new DOMParser();
         const dom: Document = domParser.parseFromString(this.emailHtmlBody, 'text/html');
+        const imgs = dom.body.querySelectorAll('img');
+        const postAttachments = new Attachments(this.attachments, this.tempFolderPath);
+
+        for (let i = 0; i < imgs.length; i++) {
+            const img = imgs[i];
 
-        dom.body.querySelectorAll('img').forEach((img)=>{
             // replace with inline attachment.
             if (/^cid:/.test(img.src)) {
                 const cid = img.src.substr(4).trim();
-                const attachment = attachmentsProp.find((attachment)=>attachment.contentId && attachment.contentId === `<${cid}>`);
+                const attachment = this.attachments.find((attachment)=>attachment.contentId && attachment.contentId === `<${cid}>`);
+
                 if (attachment) {
-                    img.src = `:/${attachment.id}`;
+                    const resource = await postAttachments.postAttachment(attachment);
+                    img.src = `:/${resource.id}`;
                 }
             }
-        });
+        };
         this.emailHtmlBody = dom.body.innerHTML;
     }
-}
 
+    async addAttachments(attachmentsStyle: string): Promise<string> {
```

Comment:
maybe it should be an enum instead of string?

Reply:
This function takes a parameter from the export criteria object attribute. I changed a parameter now from string to 'Table' | 'Link' to be more clear.